grakn/service/Session/Concept/Concept.py

```diff
@@ -18,610 +18,180 @@
 #
 
 
-from grakn.service.Session.util import enums
-from grakn.service.Session.util.RequestBuilder import RequestBuilder
-from grakn.exception.GraknError import GraknError 
-
+#from grakn.service.Session.Concept import BaseTypeMapping, ConceptFactory
+#from grakn.service.Session.util.ResponseReader import ResponseReader
+import BaseTypeMapping
+import ConceptFactory
 
 
 class Concept(object):
 
-    def __init__(self, concept_id, base_type, tx_service):
-        self.id = concept_id
-        self.base_type = base_type
-        self._tx_service = tx_service
-
-
-    def delete(self):
-        del_request = RequestBuilder.ConceptMethod.delete()
-        method_response = self._tx_service.run_concept_method(self.id, del_request)
-        return
-
-    def is_deleted(self):
-        retrieved = self._tx_service.get_concept(self.id)
-        return retrieved is None
+    def __init__(self, grpc_concept):
+        self.id = grpc_concept.id
 
+    def as_remote(self, tx_service):
```

Comment:
why does the user have a handle to the tx_service? Should this be the plain `tx`?

Reply:
It actually is the plain `tx` which is one of the bugs I had to fix, will rename it now.